### PR TITLE
doc clean-ups, remove old beam

### PR DESCRIPTION
**Pull request by vince62s**
Trying to:
- maintain the inline documentation according to recent changes.
btw: @flauted @bpopeters youmay have to do so for the Dataset recent changes, that would  be great.

- make the change "fast translate" becoming default and legacy becoming "deprecated" still in the code for now, but to be removed.

@sebastianGehrmann @guillaumekln are you guys ok ?
we need to move on I think.

### CODE REVIEW

**Comment by flauted** [comment]
@vince62s Will do. Should the docs follow the [Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) (but with the last blank line after the closing ``"""`` as in the [actual style guide](http://google.github.io/styleguide/pyguide.html))? That's the de facto format it seems but some places it's different.

File: onmt/translate/translator.py [hunk 1]
```diff
@@ -58,14 +58,12 @@ class Translator(object):
...
-       beam_size (int): size of beam to use
-       n_best (int): number of translations produced
-       max_length (int): maximum length output to produce
+       opt (obj): command line options
+       model_opt (obj): model options
        global_scores (:obj:`GlobalScorer`):
          object to rescore final translations
-       copy_attn (bool): use copy attention during translation
-       cuda (bool): use cuda
-       beam_trace (bool): trace beam search for debugging
```
Comment (bpopeters) [inline onmt/translate/translator.py]: so we're removing the `beam_trace` option for now?

File: onmt/decoders/transformer.py [hunk 2]
```diff
@@ -113,13 +113,15 @@ class TransformerDecoder(DecoderBase):
        d_model (int): size of the model
        heads (int): number of heads
        d_ff (int): size of the inner FF layer
+       copy_attn (bool): if using a seperate copy attention
```
Comment (bpopeters) [inline onmt/decoders/transformer.py]: separate

File: onmt/translate/translator.py (hunk 1 above)
Comment (vince62s) [reply onmt/translate/translator.py]: no, but it's not an argument, it's an attribute passed through opt.

**Comment by vince62s** [comment]
@flauted I am about to remove the old beam search but this is still here:
https://github.com/OpenNMT/OpenNMT-py/pull/1268/files#diff-b228765b5cdb90ddc4bf121dcdad5eabR542
Can we remove it ? did you test this beta param with the cov_penalty in the bearm_search ?

**Comment by flauted** [comment]
That link is just showing me the diff. Maybe because I'm on my phone.

I don't think the fast beam search has any coverage penalty integrated, if that's what you're asking about. I was planning on adding it but it's hard to test because of #1267.

**Comment by vince62s** [comment]
@guillaumekln @flauted well I can wait later today but in the meantime I just added the test if coverage_penalty is on then use the old beam.

**Pull request merged**